staging/src/k8s.io/api/coordination/v1/types.go

```diff
@@ -58,6 +73,17 @@ type LeaseSpec struct {
 	// holders.
 	// +optional
 	LeaseTransitions *int32 `json:"leaseTransitions,omitempty" protobuf:"varint,5,opt,name=leaseTransitions"`
+	// Strategy indicates the strategy for picking the leader for coordinated leader election
+	// A nil Strategy implies no coordination.
+	// (Alpha) Using this field requires the CoordinatedLeaderElection feature gate to be enabled.
+	// +featureGate=CoordinatedLeaderElection
+	// +optional
+	Strategy *CoordinatedLeaseStrategy `json:"strategy,omitempty" protobuf:"bytes,6,opt,name=strategy"`
+	// PreferredHolder signals to a lease holder that the lease has a
```

Comment:
Does this apply to ALL leases or only when Strategy is specified?

Reply:
When Strategy is specified but Strategy and PreferredHolder selection is done at the same time. CLE gathers the candidates and picks an optimal leader based on a Strategy. CLE then updates the lease object with the PreferredHolder and Strategy. PreferredHolder and Strategy are set at the same time. If there are no candidates, there is no Strategy and thus no PreferredHolder.

Reply:
> When Strategy is specified

Then spec it as such.  Otherwise it is implied that a no-strategy lease must also be relinquished if this field is set.

And maybe validate that it can only set set if strategy is set

Reply:
added.